commands/prune.go

```diff
@@ -50,7 +50,7 @@ func runPrune(ctx context.Context, dockerCli command.Cli, opts pruneOptions) err
 	}
 
 	if !opts.force {
-		if ok, err := prompt(ctx, dockerCli.In(), dockerCli.Out(), warning); err != nil {
+		if ok, err := command.PromptForConfirmation(ctx, dockerCli.In(), dockerCli.Out(), warning); err != nil {
```

Comment:
this has been fixed in https://github.com/docker/cli/pull/4888

Reply:
The current version can't be used because it sets up own signal handlers internally that would mess up the buildx own signal handlers. See https://github.com/docker/cli/pull/4888#pullrequestreview-1937955718

We need to do some copy-paste of the old/forked code to work around.